server/server.go

```diff
@@ -166,24 +169,33 @@ func Main(keyP crypki.KeyIDProcessor) {
 	recoveryHandler := func(p interface{}) (err error) {
 		return status.Errorf(codes.Internal, "internal server error")
 	}
+
+	var server *http.Server
+	interceptors := []grpc.UnaryServerInterceptor{
+		recovery.UnaryServerInterceptor(recovery.WithRecoveryHandler(recoveryHandler)),
+	}
+	if cfg.ShutdownOnFrequentSigningFailure {
+		interceptors = append(interceptors, StatusInterceptor((newShutdownCounter(func() {
+			if err := server.Shutdown(ctx); err != nil {
+				log.Fatalf("failed to shutdown server: %v", err)
+			}
+		})).Interceptor))
+	}
```

Comment:
From the documentation of [ChainUnaryInterceptor](https://godoc.org/google.golang.org/grpc#ChainUnaryInterceptor):
> The first interceptor will be the outer most, while the last interceptor will be the inner most wrapper around the real call.

As a result, we may need to change the order of the two interceptors here. Otherwise, if a panic occurs in the actual handler, the counter won't be updated because anything below [here](https://github.com/theparanoids/crypki/pull/45/files#diff-913b93024b5e160034b82e8ff754a663R15) won't be executed, and the panic will only be recovered after returning from the counter-shutdown interceptor, namely in the recovery interceptor.